app/src/main/res/layout/activity_detail.xml

```diff
@@ -51,6 +51,7 @@
                 android:id="@+id/toolbar"
                 android:layout_width="match_parent"
                 android:layout_height="?attr/actionBarSize"
+                android:fitsSystemWindows="true"
```

Comment:
private void fixApi21ToolBarBug(Toolbar toolbar){ //bug that puts toolbar out of screen
        if(Build.VERSION.SDK_INT!=21) return; //only on api 21
        final int resourceId = getResources().getIdentifier("status_bar_height", "dimen", "android");
        final int result = (resourceId>0) ? getResources().getDimensionPixelSize(resourceId)*2 : 0;
        final CollapsingToolbarLayout.LayoutParams params = 
                (CollapsingToolbarLayout.LayoutParams)toolbar.getLayoutParams();
        params.topMargin -= result;
        toolbar.setLayoutParams(params);
    }